### ISSUE TITLE
allow style of pointer to be changed

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#2474]
When I mouse over a link, I want a little icon change that makes it clear I can click on it. Right now, there's no way for a program to provide that kind of visual pointer feedback. Add it, when supported.

**Comment by klamonte** [comment]
It would be pretty cool if there was a sequence to pick the right cursor from some union subset of the [X11 cursors](https://tronche.com/gui/x/xlib/appendix/b/) and the [Win32 cursors](https://docs.microsoft.com/en-us/windows/win32/menurc/about-cursors).  Or maybe take one of the image formats and pass a cursor icon via OSC or similar.

**Comment by j4james** [comment]
FYI, XTerm uses `OSC 22` for setting the mouse cursor shape and so does Mintty, but they support completely different sets of cursor names (I'm assuming XTerm is using the X11 names, and Mintty is using Win32).

**Comment by klamonte** [comment]
Oh wow OSC 22 is pretty new: Patch #367 - 2021/03/26

**Comment by dnkl** [comment]
Doesn't look like applications have a way to reset the cursor?

**Comment by j4james** [comment]
On both Mintty and XTerm you can reset it by giving a cursor name that's not recognised, e.g.

`printf "\e]22;putitbacktowhatitwas\e\\"`

Mintty also accepts a blank name as a way to reset, which would have been a good solution, but that doesn't seem to work on XTerm.

**Comment by dnkl** [comment]
@j4james thanks! Though I'm slightly bothered by the fact that it's not documented. Might add support to foot. Seems easy enough at least...

**Comment by klamonte** [comment]
I'm miffed that xterm did not bother to support a "no cursor" cursor here. There is still no way afaik to completely turn off the mouse pointer.

**Comment by dnkl** [comment]
Yeah, having the empty string mean "no cursor", and then maybe add OSC-122 to reset doesn't seem too far fetched to me...

**Comment by dankamongmen** [comment]
changed issue name in the hope of no longer looking at this and wondering "what the fuck does that mean?"
